### PR TITLE
Multiselect issue in project module

### LINKED ISSUE #189: Team members are not displayed while editing project
While editing a project. The drop down displaying the list of members of the project do not get populated. Please resolve it

### PR DESCRIPTION
**Pull request by iamronakshah**
Fixed #189 

Files Changed:
 ProjectRepository.cs - use mapper , updated comments.
IProjectRepository.cs - updated comments.
ProjectTests.cs - Updated test cases.
ProjectController.cs - Removed unnecessary Exception.
idsvr3test.pfx - added idsvr3test.pfx file.  

 **Not Review Files**
project-edit.component.ts - updated logic for binding data to md2 multi select **(Do Not Review, Will be fixed in #197 )**
project-view.component.ts - updated logic for binding data to md2 multi select **(Do Not Review, Will be fixed in #197 )**
project-list.component.ts - Added DatePipe **(Do Not Review, Will be fixed in #197 )**
project.model.ts - Added createdOns and updatedOns. **(Do Not Review, Will be fixed in future pull request)**  
 project.service.ts - resolve tslint issue. **(Do Not Review, Will be fixed in #197 )**

### CODE REVIEW

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ProjectsRepository/ProjectRepository.cs [hunk 1]
```diff
@@ -86,70 +68,67 @@ public class ProjectRepository : IProjectRepository
...
+        /// This method to return the project details of the given id 
         /// </summary>
-        /// <param name="id"></param>Project id that need to be fetch the Project and list of users
-        /// <returns></returns>Project and User/Users infromation 
-        /// 
+        /// <param name="id">project id</param>
+        /// <returns>project infromation </returns>
         public async Task<ProjectAc> GetProjectByIdAsync(int id)
         {
             List<UserAc> userAcList = new List<UserAc>();
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ProjectsRepository/ProjectRepository.cs]: This is useless assignment. no need to new this class here.

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ProjectsRepository/ProjectRepository.cs [hunk 2]
```diff
@@ -191,171 +170,152 @@ public class ProjectRepository : IProjectRepository
...
-                    projectAc.TeamLeaderId = project.TeamLeaderId;
-                    project.UpdatedBy = project.UpdatedBy;
-                }
-                if (projectAc==null)
-                    throw new ProjectNotFound();
-                else
-                    return projectAc;
-          }
+            Project project = await _projectDataRepository.FirstOrDefaultAsync(x => x.SlackChannelName == slackChannelName);
+            ProjectAc projectAc = new ProjectAc();
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ProjectsRepository/ProjectRepository.cs]: this is again useless assignment

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ProjectsRepository/ProjectRepository.cs [hunk 3]
```diff
@@ -191,171 +170,152 @@ public class ProjectRepository : IProjectRepository
...
+        /// <param name="userId">passed login user id</param>
+        /// <returns>project information</returns>
         public async Task<IEnumerable<ProjectAc>> GetAllProjectForUserAsync(string userId)
         {
-            var projects =(await _projectDataRepository.FetchAsync(x => x.TeamLeaderId == userId)).ToList();
-            _logger.LogInformation("Total Projects " + projects.Count().ToString()); 
-            var projectUser =(await _projectUserDataRepository.FetchAsync(x => x.UserId == userId)).ToList();
-            _logger.LogInformation("Total UserProjects " + projectUser.Count().ToString());
             List<ProjectAc> projectAcList = new List<ProjectAc>();
             ProjectAc projectAc = new ProjectAc();
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ProjectsRepository/ProjectRepository.cs]: This also is not useful

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ProjectsRepository/ProjectRepository.cs [hunk 4]
```diff
@@ -191,171 +170,152 @@ public class ProjectRepository : IProjectRepository
...
                     projectAc.TeamLeader = projectMapper;
-                    projectAc.IsActive = project.IsActive;
                     projectAcList.Add(projectAc);
                 }
             }
+            //list of project where user are as team member.
+            var projectUser = await _projectUserDataRepository.Fetch(x => x.UserId == userId).ToListAsync();
             foreach (var project in projectUser)
             {
                 projectAc = new ProjectAc();
```
Comment (chintans) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ProjectsRepository/ProjectRepository.cs]: Here too

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ProjectsRepository/ProjectRepository.cs (hunk 1 above)
Comment (iamronakshah) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ProjectsRepository/ProjectRepository.cs]: removed useless assignment.

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ProjectsRepository/ProjectRepository.cs (hunk 2 above)
Comment (iamronakshah) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ProjectsRepository/ProjectRepository.cs]: removed useless assignment.

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ProjectsRepository/ProjectRepository.cs (hunk 3 above)
Comment (iamronakshah) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ProjectsRepository/ProjectRepository.cs]: removed useless assignment.

File: Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ProjectsRepository/ProjectRepository.cs (hunk 4 above)
Comment (iamronakshah) [inline Promact.Oauth.Server/src/Promact.Oauth.Server/Repository/ProjectsRepository/ProjectRepository.cs]: removed useless assignment.

**Pull request merged**